src/DotLiquid/StandardFilters.cs

```diff
@@ -365,12 +393,7 @@ public static string Currency(Context context, object input, string languageTag
             var culture = languageTag == null ? context.CurrentCulture : new CultureInfo(languageTag.Trim());
 
             // Attempt to convert to a currency using the context current culture.
-            if (IsReal(input))
-                return Convert.ToDecimal(input).ToString("C", culture);
-            if (decimal.TryParse(input.ToString(), NumberStyles.Currency, context.CurrentCulture, out decimal amount))
-                return amount.ToString("C", culture);
-
-            return input.ToString();
+            return TryParseDecimal(context.CurrentCulture, input, out var d) ? d.ToString("C", culture) : input.ToString();

```

Comment:
Note to reviewer: this doesn't make sense to me but is kept as-is to avoid breaking tests.

Reply:
@robin-parker maybe you want to chime in on the currency filter behavior?